packages/core/test/schema.test.ts

```diff
@@ -1,4 +1,4 @@
-import { createMachine, createSchema } from '../src/index.ts';
+import { createMachine } from '../src/index.ts';
```

Comment:
It feels like this test file could be removed. It doesn't quite test anything new/useful. Other tests are already testing inference based on `.types` - the `fromJSONSchema` thing here is completely separate from XState, it doesn't matter for inference anyhow (only its return type matters but we already test that elsewhere). 

The "schema should be present in the machine definition" test is also not testing something that we should be guarantee. The `.types` purpose is only type-level and runtime-wise its quite irrelevant. In fact, in this test - its runtime type doesnt match the type-level one, so this is pretty much unusable without casting to `any` or something.